packages/tunnel-client/src/lib/tunnel.ts

```diff
@@ -333,6 +333,69 @@ export class Tunnel extends (EventEmitter as new () => TypedEmitter<TunnelEvents
     this.emit("close");
   }
 
+  private async createProxyConnection(
+    proxyUrl: string,
+    targetHost: string,
+    targetPort: number,
+    useTls: boolean,
+  ): Promise<net.Socket> {
+    const url = new URL(proxyUrl);
+    const proxyHost = url.hostname;
+    const proxyPort =
+      parseInt(url.port) || (url.protocol === "https:" ? 443 : 80);
+
+    const proxySocket = net.connect({
```

Comment:
This won't work for HTTPS proxies although the logic above would suggest otherwise. Maybe we handle it correctly here or just throw?